inference-engine/src/inference_engine/include/openvino/runtime/remote_context.hpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for the IE RemoteContext and RemoteBlob classes
+ *
+ * @file ie_remote_context.hpp
+ */
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "common.hpp"
+#include "details/ie_so_loader.h"
+#include "ie_parameter.hpp"
+#include "ie_remote_context.hpp"
+
+namespace InferenceEngine {
+class IRemoteContext;
+class RemoteBlob;
+}  // namespace InferenceEngine
+
+namespace ov {
+namespace runtime {
+
+class Core;
+
+/**
+ * @brief This class represents an abstraction
+ * for remote (non-CPU) accelerator device-specific execution context.
+ * Such context represents a scope on the device within which executable
+ * networks and remote memory blobs can exist, function and exchange data.
+ */
+class INFERENCE_ENGINE_API_CLASS(RemoteContext) {
+    ie::details::SharedObjectLoader _so;
+    std::shared_ptr<ie::IRemoteContext> _impl;
```

Comment:
Can we avoid introducing structures from legacy namespace?
For Core we have:
```
class Impl;
std::shared_ptr<Impl> _impl;
```
without introducing internal things.

Reply:
The `Core` is not a part of plugin API. But IRemoteContext is. 
`ov::RemoteContext` - is a part of public c++ API. It is just a wrapper over internal `ie::IRemoteContext`. 
A plugin creates `ie::IRemoteContext` instance and populates this wrapper. `ie::IRemoteContext` inherits from `ie::RemoteContext` to preserve backward compatibility with the current plugin API.